### PR TITLE
#168131437 Travel Admins should be able to create Accommodation facilities

### PR DESCRIPTION
**Pull request by Bahatiroben**
#### What does this PR do?
- It allows Travel administrator to create accommodation facilities and add rooms to them.
#### Description of Task to be completed?
- Given that the user is a travel admin: 
```
he should be allowed to add:
- accommodations
- rooms
```
##### Adding accommodation POST `api/v1/accommodations`
- payload: 
```
{
    "name": "name1",
    "locationId": 1
}
```

##### Adding rooms POST `api/v1/accommodations/rooms`
```
{
"name": "room",
"type": "2 bedroom",
"accommodationId": 1
}
```
 
#### How should this be manually tested?
```
- clone the repo
- run `npm install` to install dependencies
- add required environment variables
- run npm run start:dev
```
- signin with these Travel administrator credentials
``` 
  userPassword: 'Root1123#'
``` 
```
  userEmail: 'bahatiroben@gmail.com',
```
- create the accommodations and rooms using the endpoints and payloads specified above
#### Any background context you want to provide?
-  I have included a new table of Locations hence new migration and model for it.
- I have used cloudinary for picture upload hence the environment variables used are in .env.example
- to reflect the new tables run `npm db-migrate:dev`
- I have seeded the Travel Administrator user and Locations
- seeds can be run by running `npm run db-seed:test`

- the credentials of the travel Admin are used for testing.

#### What are the relevant pivotal tracker stories?
[#168131437](https://www.pivotaltracker.com/story/show/168131437)
#### Screenshots (if appropriate)

image
image

#### Questions:
N/A

### CODE REVIEW

**Comment by Dkabiswa** [comment]
nice work but can you shade more light on how you run the seed files and shre the text passwords you used 
also, I see new migration files how do I add them to my database?
Which environment variables should I add?
I also see locations files can you maybe explain in background content

**Comment by Bahatiroben** [comment]
> nice work but can you shade more light on how you run the seed files and shre the text passwords you used
> also, I see new migration files how do I add them to my database?
> Which environment variables should I add?
> I also see locations files can you maybe explain in background content

sure, I am going to add it in the descriptions of PR.

**Comment by drayzii** [comment]
I have seen you didn't test cloudinary. You can use `sinon` to test it.

**Comment by drayzii** [comment]
There are also some validations you didn't test

File: src/validation/accommodationValidation.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+   * @returns {object}.
+   */
+  static async validateCreateAccommodation(req, res, next) {
+    const schema = Joi.object().keys({
+      name: Joi.string()
+        .min(2)
+        .trim()
+        .required()
+        .error(() => 'name field is required and must be a string'),
+      locationId: Joi.number()
```
Comment (Dkabiswa) [inline src/validation/accommodationValidation.js:25]: you can add .integer to exclude floating-point numbers but also the error message directs a user to enter a string check it out  
Also the key is locationId yet error message indicates location

**Review by Dkabiswa — COMMENTED** [review]
check it out

**Comment by drayzii** [comment]
Can you change seeders and migrations to ES6

**Comment by drayzii** [comment]
image
It is returning `409` for location not found

**Comment by drayzii** [comment]
```
{
    "status": 400,
    "message": "validations failed",
    "error": [
        {
            "message": "location is required and must be a string",
            "path": [
                "locationId"
            ],
            "type": "number.base",
            "context": {
                "value": "dftghj",
                "key": "locationId",
                "label": "locationId"
            }
        }
    ]
}
```
Wrong validation message, and the status has to be `422`

**Comment by Bahatiroben** [comment]
> ```
> {
>     "status": 400,
>     "message": "validations failed",
>     "error": [
>         {
>             "message": "location is required and must be a string",
>             "path": [
>                 "locationId"
>             ],
>             "type": "number.base",
>             "context": {
>                 "value": "dftghj",
>                 "key": "locationId",
>                 "label": "locationId"
>             }
>         }
>     ]
> }
> ```
> 
> Wrong validation message, and the status has to be `422`

changed

**Comment by Bahatiroben** [comment]
> ```
> {
>     "status": 400,
>     "message": "validations failed",
>     "error": [
>         {
>             "message": "location is required and must be a string",
>             "path": [
>                 "locationId"
>             ],
>             "type": "number.base",
>             "context": {
>                 "value": "dftghj",
>                 "key": "locationId",
>                 "label": "locationId"
>             }
>         }
>     ]
> }
> ```
> 
> Wrong validation message, and the status has to be `422`

changed

**Review by drayzii — APPROVED** [review]
LGTM

**Review by Dkabiswa — APPROVED** [review]

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**